### PR TITLE
[mso_schema_clone] Supports cloning the schema issue #190

### PR DESCRIPTION
**Pull request by anvitha-jain**
solves #190

### CODE REVIEW

File: plugins/modules/mso_schema_clone.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    type: str
+    aliases: [ name, displayName ]
+  state:
+    description:
+    - Use C(clone) for adding.
+    type: str
+    choices: [ clone ]
+    default: clone
+notes:
+- Due to restrictions of the MSO REST API this module cannot create empty schemas (i.e. schemas without templates).
```
Comment (lhercot) [inline plugins/modules/mso_schema_clone.py:41]: I don't think this note is needed here.

File: plugins/modules/mso_schema_clone.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  Use the M(cisco.mso.mso_schema_template) to automatically create schemas with templates.
```
Comment (lhercot) [inline plugins/modules/mso_schema_clone.py:42]: Same here

File: plugins/modules/mso_schema_clone.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+options:
+  source_schema:
+    description:
+    - The name of the source_schema.
+    type: str
+  destination_schema:
+    description:
+    - The name of the destination_schema.
+    type: str
+    aliases: [ name, displayName ]
```
Comment (lhercot) [inline plugins/modules/mso_schema_clone.py:33]: remove aliases. I don't think they are clear enough.

File: plugins/modules/mso_schema_clone.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    destination_schema_id = None
+    path = 'schemas'
+
+    # Check if source and destination schema are named differently
+    if source_schema == destination_schema:
+        mso.fail_json(msg="Schema with the name '{0}' already exists. Please use another name.".format(destination_schema))
+    # Query for existing object(s)
+    if destination_schema:
+        mso.existing = mso.get_obj(path, displayName=destination_schema)
+        if mso.existing:
```
Comment (lhercot) [inline plugins/modules/mso_schema_clone.py:96]: Does the UI allow you to clone to a destination schema that already exists?

File: plugins/modules/mso_schema_clone.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    destination_schema = module.params.get('destination_schema')
+    state = module.params.get('state')
+
+    mso = MSOModule(module)
+
+    # Get source schema details
+    source_schema_path = "schemas/{0}".format(mso.lookup_schema(source_schema))
+    source_schema_obj = mso.query_obj(source_schema_path, displayName=source_schema)
+
+    source_data = source_schema_obj.get('templates')
```
Comment (lhercot) [inline plugins/modules/mso_schema_clone.py:85]: You don't seems to fail if source_schema does not exist. Is that normal?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: plugins/modules/mso_schema_clone.py (hunk 4 above)
Comment (anvitha-jain) [reply plugins/modules/mso_schema_clone.py:96]: I check this condition in the line above.

**Review by lhercot — APPROVED** [review]

**Pull request merged**